src/QuickMarcEditor/QuickMarcEditorRows/FixedField/FixedFieldFactory.test.js

```diff
@@ -4,85 +4,120 @@ import { FixedFieldFactory } from './FixedFieldFactory';
+      FixedFieldFactory.getFixedField(intl, 'records', fixedFieldSpecBib, 'a', 'i').props.config.type,
     ).toBe('continuing_resources');
 
     expect(
-      FixedFieldFactory.getFixedField('records', fixedFieldSpecBib, 'a', 's').props.config.type,
+      FixedFieldFactory.getFixedField(intl, 'records', fixedFieldSpecBib, 'a', 's').props.config.type,
     ).toBe('continuing_resources');
 
     expect(
-      FixedFieldFactory.getFixedField('records', fixedFieldSpecBib, 's').props.config.type,
+      FixedFieldFactory.getFixedField(intl, 'records', fixedFieldSpecBib, 's').props.config.type,
     ).toBe('continuing_resources');
 
     expect(
-      FixedFieldFactory.getFixedField('records', fixedFieldSpecBib, 'm').props.config.type,
+      FixedFieldFactory.getFixedField(intl, 'records', fixedFieldSpecBib, 'm').props.config.type,
     ).toBe('computer_files');
 
     expect(
-      FixedFieldFactory.getFixedField('records', fixedFieldSpecBib, 'g').props.config.type,
+      FixedFieldFactory.getFixedField(intl, 'records', fixedFieldSpecBib, 'g').props.config.type,
     ).toBe('visual_materials');
 
     expect(
-      FixedFieldFactory.getFixedField('records', fixedFieldSpecBib, 'e').props.config.type,
+      FixedFieldFactory.getFixedField(intl, 'records', fixedFieldSpecBib, 'e').props.config.type,
     ).toBe('maps');
 
     expect(
-      FixedFieldFactory.getFixedField('records', fixedFieldSpecBib, 'f').props.config.type,
+      FixedFieldFactory.getFixedField(intl, 'records', fixedFieldSpecBib, 'f').props.config.type,
     ).toBe('maps');
 
     expect(
-      FixedFieldFactory.getFixedField('records', fixedFieldSpecBib, 'p').props.config.type,
+      FixedFieldFactory.getFixedField(intl, 'records', fixedFieldSpecBib, 'p').props.config.type,
     ).toBe('mixed_materials');
 
     expect(
-      FixedFieldFactory.getFixedField('records', fixedFieldSpecAuth, 'z').props.config.type,
+      FixedFieldFactory.getFixedField(intl, 'records', fixedFieldSpecAuth, 'z').props.config.type,
     ).toBe('unknown');
 
     expect(
-      FixedFieldFactory.getFixedField('records', fixedFieldSpecHold, 'u').props.config.type,
+      FixedFieldFactory.getFixedField(intl, 'records', fixedFieldSpecHold, 'u').props.config.type,
     ).toBe('unknown');
   });
 
   it('should return undefined type when there is no matched field', () => {
     expect(
-      FixedFieldFactory.getFixedField('records', fixedFieldSpecBib, 'l').props.config.type,
+      FixedFieldFactory.getFixedField(intl, 'records', fixedFieldSpecBib, 'l').props.config.type,
     ).toBe(undefined);
   });
+
+  it('should field not to be in config when is ReadOnly', () => {
+    const fields = FixedFieldFactory.getFixedField(intl, 'records', fixedFieldSpecBib, 'a', 'm').props.config.fields;
+    const fieldsEntered = fields.filter(x => x.name === 'Entered');
+
+    expect(fieldsEntered).toHaveLength(0);
+  });
+
+  it('for type books should return type "String" for field Date1', () => {
```

Comment:
please format tests in such a way that conditions are written in `describe`
For example,
```
describe('when type is books', () => {
  it('should return type "String" for field Date1', () => {...});
});
```